icu4c/source/i18n/unicode/msgfmt.h

```diff
@@ -245,8 +245,9 @@ class NumberFormat;
  *
  * <h4>Argument formatting</h4>
  *
- * <p>Arguments are formatted according to their type, using the <code>toString()</code>
- * methods. But there are a few types that are known to be locale-sensitive.
+ * <p>Arguments are formatted according to their type, using the using default
+ * <code>Formattable</code> for those types, unless otherwise specified.
```

Comment:
Also, I just realized that in Java we should fix the equivalent sentence there too.
While it uses toString() if it does not recognize the type, it does use the default formatters for certain types.

GitHub won't let me comment on that line in the .java file, so here is my suggestion for that (including unchanged text):

Arguments are formatted according to their type, using the using default formatters for those types, unless otherwise specified. But there are a few types ... These include numbers, dates, times, and durations. For other types, toString() is used.